hack/versions/cmd/new/version.go

```diff
@@ -76,13 +76,13 @@ func main() {
 	hackschema.UpdateVersionComment(path("latest", "config.go"), false)
 
 	// Update skaffold.yaml in integration tests
-	walk.From("integration").WhenHasName("skaffold.yaml").MustDo(func(path string, _ walk.Dirent) error {
+	walk.From("integration").WhenNameContains("skaffold", "yaml").MustDo(func(path string, _ walk.Dirent) error {
```

Comment:
just for the sake of argument, if we added a `generate_skaffold_yaml.sh` someday it'd match this filter. I wonder if we should add a `IsFileType(extension string)` filter and do 
```go
walk.From("integration").IsFileType("yaml").WhenNameContains("skaffold").MustDo(...)
```

Reply:
good point, I can add this as well

Reply:
Would it make sense to have `WhenNameMatches(glob)` and use [`path.Match()`](https://pkg.go.dev/path/#Match)?

Reply:
@briandealwis I think that'd make sense too. It would definitely shorten things a bit as well. Do you think we should go with that? I just pushed changes if you want to take a look and let me know

Reply:
I would just have keep it as simple as possible: since `WhenNameContains` and `IsFileType()` can be implemented with `WhenNameMatches()`, I would just keep `WhenNameMatches()`.

(My quibble with `IsFileType()` is that it could imply doing content-matching.)

Reply:
Makes sense :) I'll make changes in favor of `WhenNameMatches()`